File: java/utility/PreparePath.java
package voxgig.bluefindecryptxsdk.utility;

import java.util.ArrayList;
import java.util.List;

import voxgig.bluefindecryptxsdk.core.Context;
import voxgig.bluefindecryptxsdk.utility.struct.Struct;

@SuppressWarnings({"unchecked"})
final class PreparePath {

  private PreparePath() {}

  static String preparePath(Context ctx) {
    List<Object> parts = null;
    Object p = Struct.getprop(ctx.point, "parts");
    if (p instanceof List) {
      parts = (List<Object>) p;
    }
    if (parts == null) {
      parts = new ArrayList<>();
    }

    return Struct.join(parts, "/", true);
  }
}

File: php/utility/PreparePath.php
<?php
declare(strict_types=1);

// BluefinDecryptx SDK utility: prepare_path

class BluefinDecryptxPreparePath
{
    public static function call(BluefinDecryptxContext $ctx): string
    {
        $point = $ctx->point;
        $parts = [];
        if ($point) {
            $p = \Voxgig\Struct\Struct::getprop($point, 'parts');
            if (is_array($p)) {
                $parts = $p;
            }
        }
        return \Voxgig\Struct\Struct::join($parts, '/', true);
    }
}
